simpleflow/execute.py

```diff
@@ -181,7 +183,21 @@ def execute(*args, **kwargs):
                     close_fds=close_fds,
                     pass_fds=pass_fds,
                 )
-                rc = process.wait()
+
+                # Wait for process
+                if timeout:
+                    t_start = time.time()
+                    rc = process.poll()
+                    while time.time() - t_start < timeout and rc is None:
+                        time.sleep(1)
+                        rc = process.poll()
+
+                    if rc is None:
+                        process.terminate()  # send SIGTERM
```

Comment:
Strictly speaking this could raise a `OSError(3, 'No such process')` because you have no guarantee that the subprocess won't terminate between the previous `process.poll()` and the execution of this line. try/except and check for `exception.errno == 3` ?